src/consensus/aft/raft.h

```diff
@@ -171,13 +173,17 @@ namespace aft
       std::chrono::milliseconds view_change_timeout_,
       size_t sig_tx_interval_ = 0,
       bool public_only_ = false,
-      kv::ReplicaState initial_state_ = kv::ReplicaState::Follower,
+      std::optional<kv::LeadershipState> initial_leadership_state_ =
+        kv::LeadershipState::Follower,
+      kv::MembershipState initial_membership_state_ =
+        kv::MembershipState::Active,
       ReconfigurationType reconfiguration_type_ =
         ReconfigurationType::ONE_TRANSACTION) :
       consensus_type(consensus_type_),
       store(std::move(store_)),
 
-      replica_state(initial_state_),
+      leadership_state(initial_leadership_state_),
```

Comment:
Shouldn't our initial leadership state be unset, and only be set once we transition to Active? I thought this was the whole reason it was optional.

Reply:
Yes, for 2tx that's the case (`node_state.h:1976`), but for 1tx I kept the previous default. You're right though, I'll change that and check nothing breaks.